### PR TITLE
[now-cli] Fix `now dev` when api file is added or removed

### PR DESCRIPTION
**Pull request by styfle**
This PR is a follow up to #3703 which fixes file output renaming on initial boot but not when files are added or removed while `now dev` is running. This PR fixes that behavior.

### CODE REVIEW

**Comment by peadrakw** [comment]
now.json

File: packages/now-cli/src/util/dev/server.ts [hunk 1]
```diff
@@ -1130,6 +1140,19 @@ export default class DevServer {
     }
   }
 
+  getExtensionlessFile = (path: string) => {
+    const ext = extname(path);
+    if (
+      this.apiDir &&
+      path.startsWith(this.apiDir + '/') &&
```
Comment (TooTallNate) [inline packages/now-cli/src/util/dev/server.ts:1147]: Will this be correct on Windows?

File: packages/now-cli/src/util/dev/server.ts [hunk 2]
```diff
@@ -1130,6 +1140,19 @@ export default class DevServer {
     }
   }
 
+  getExtensionlessFile = (path: string) => {
```
Comment (TooTallNate) [inline packages/now-cli/src/util/dev/server.ts:1143]: Does the function really need to be bound to `this`?

```suggestion
  getExtensionlessFile(path: string): string | null {
```

**Review by TooTallNate — APPROVED** [review]
A test case would be ideal, but I guess adding one would be tricky since a file needs to be created after the server has booted up 🤔

File: packages/now-cli/src/util/dev/server.ts (hunk 2 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/server.ts:1143]: It save us passing several variables that are already bound to `this` (dev server). We already store `this.nowConfig` so I stored the stuff I needed there too.

File: packages/now-cli/src/util/dev/server.ts (hunk 1 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/server.ts:1147]: I guess I could add windows to the `now dev` CI 🤔

**Review by AndyBitz — APPROVED** [review]

File: packages/now-cli/src/util/dev/server.ts (hunk 1 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/server.ts:1147]: I confirmed this works with Windows. I also created PR #3727 to add Windows to CI.

**Pull request merged**